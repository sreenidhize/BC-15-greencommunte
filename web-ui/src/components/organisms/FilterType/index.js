import React from 'react';
import Filter from '../../molecules/Filter/Filter';
import {makeStyles} from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
    heading: {
        fontFamily: theme.typography.fontFamily,
        fontWeight: 600,
        fontSize: 14,
        marginBottom: -2,
    },
    fliterLabel: {
        marginBottom: -10,
    },
}));

const FilterType = (props) => {
    const classes = useStyles();
    const {labels, heading, handleChange, filters} = props;
    const Filters = labels.map((label) => {
        return (
            <div>
                {
                    label.option && 
                    <div className={classes.fliterLabel}>
                        <Filter filters={filters} label={label.option} handleChange={handleChange}/>
                    </div>
                }
            </div>);
    });
    return (
        <div data-testid="filter-type">
            <div className={classes.heading}>{heading}</div>
            {Filters}
        </div>
    );
};

export default FilterType;