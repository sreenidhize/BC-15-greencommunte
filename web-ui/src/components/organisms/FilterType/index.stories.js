import React from "react";
import FilterType from ".";
import {labels} from '../../../utils/constants/constant';

export default {
  title: "organisms/FilterType",
  component: FilterType,
};

const Template = (args) => <FilterType {...args} />;

export const Primary = Template.bind({});
Primary.args = {
    heading: "Distance",
    labels: labels
  };