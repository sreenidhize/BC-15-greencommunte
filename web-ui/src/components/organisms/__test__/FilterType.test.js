import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import FilterType from "../FilterType";

const heading = "Distance";
const labels = [
  {
    option: "0-10 kms",
    selected: false,
  },
  {
    option: "10-20 kms",
    selected: false,
  },
  {
    option: "20-30 kms",
    selected: false,
  },
  {
    option: "30-40 kms",
    selected: false,
  }
];

describe("Filter Type placeholder test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <FilterType labels={labels} heading={heading} filters={[]}/>
    );
    expect(container).toMatchSnapshot();
  });

  test("test filter-type render", () => {
    const { getByTestId } = render(<FilterType labels={labels} heading={heading} filters={[]} />);
    expect(getByTestId("filter-type")).toBeInTheDocument();
  });
});