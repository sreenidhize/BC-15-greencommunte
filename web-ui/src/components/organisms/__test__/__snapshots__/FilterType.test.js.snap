// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Type placeholder test Matches the snapshot 1`] = `
<div>
  <div
    data-testid="filter-type"
  >
    <div
      class="makeStyles-heading-1"
    >
      Distance
    </div>
    <div>
      <div
        class="makeStyles-fliterLabel-2"
      >
        <label
          class="MuiFormControlLabel-root"
          data-testid="filter"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root Component-root-4"
            data-testid="filter-checkbox"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-9"
                data-indeterminate="false"
                name="0-10 kms"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <span
              class="makeStyles-checkboxLabel-3"
            >
              0-10 kms
            </span>
          </span>
        </label>
      </div>
    </div>
    <div>
      <div
        class="makeStyles-fliterLabel-2"
      >
        <label
          class="MuiFormControlLabel-root"
          data-testid="filter"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root Component-root-4"
            data-testid="filter-checkbox"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-9"
                data-indeterminate="false"
                name="10-20 kms"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <span
              class="makeStyles-checkboxLabel-3"
            >
              10-20 kms
            </span>
          </span>
        </label>
      </div>
    </div>
    <div>
      <div
        class="makeStyles-fliterLabel-2"
      >
        <label
          class="MuiFormControlLabel-root"
          data-testid="filter"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root Component-root-4"
            data-testid="filter-checkbox"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-9"
                data-indeterminate="false"
                name="20-30 kms"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <span
              class="makeStyles-checkboxLabel-3"
            >
              20-30 kms
            </span>
          </span>
        </label>
      </div>
    </div>
    <div>
      <div
        class="makeStyles-fliterLabel-2"
      >
        <label
          class="MuiFormControlLabel-root"
          data-testid="filter"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root Component-root-4"
            data-testid="filter-checkbox"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-9"
                data-indeterminate="false"
                name="30-40 kms"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <span
              class="makeStyles-checkboxLabel-3"
            >
              30-40 kms
            </span>
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
